examples/getting_started/react/components/AddTodo.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useRef } from 'react';
+
+const AddTodo = ({ addTodo, cancel }) => {
+
+  const titleRef = useRef();
+  const descriptionRef = useRef();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+   
+    addTodo({
```

Comment:
Question: is there any strong reason that we pass in the `addTodo` mutation instead of creating/defining it in here? Defining it in here kinda encapsulates the entire creation logic within the component itself and it also has the potential benefit for the reader/user that they could be pointed to this file and see a really tidy, complete example of performing a mutation. What do you think?

Reply:
I kinda liked that because it will be nice way to reference that from the docs - zero react stuff just pure offix logic, but no strong opinion

Reply:
Also I see this pattern in couple production apps (and urigo react trainings)

Reply:
Fair enough, let's keep it as is :)

Reply:
I agree about encapsulating all the creation logic. I was getting a memory leak error when I did that, because the addTodo creates a promise, if the app goes offline, the promise isn't resolved and when the component unmounts, React gives out a memory leak error. Maybe I could get your eyes on this to try and help debug